interfaces/console/src/app/console/sites/page.tsx

```diff
@@ -526,19 +274,81 @@ export default function Page() {
       },
     });
 
+    if (sitesList.length > 0) {
+      fetchSiteMetrics();
+    }
+
     return () => {
       sitesList.forEach((site) => {
-        const siteKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Site}-${site.id}`;
-        const batteryKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Battery}-${site.id}`;
-        const backhaulKey = `stat-${user.orgName}-${user.id}-${Stats_Type.MainBackhaul}-${site.id}`;
-
-        PubSub.unsubscribe(siteKey);
-        PubSub.unsubscribe(batteryKey);
-        PubSub.unsubscribe(backhaulKey);
+        const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Site}-${site.id}`;
+        PubSub.unsubscribe(sKey);
       });
     };
   }, [sitesList]);
 
+  useEffect(() => {
+    const setupSubscriptions = async () => {
+      const to = getUnixTime();
+      const from = to - 30; // Last 30 seconds
+
+      // Clear existing subscriptions
+      sitesList.forEach((site) => {
+        const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Site}-${site.id}`;
+        PubSub.unsubscribe(sKey);
+      });
+
+      // Set up new subscriptions for each site
+      sitesList.forEach((site) => {
+        const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.Site}-${site.id}`;
+
+        // Initial subscription
+        MetricStatBySiteSubscription({
+          key: sKey,
+          siteId: site.id,
+          userId: user.id,
+          url: env.METRIC_URL,
+          orgName: user.orgName,
+          type: Stats_Type.Site,
+          from,
+        });
+
+        // Subscribe to updates
+        PubSub.subscribe(sKey, handleSiteStatSubscription);
+
+        // Set up interval for periodic updates
+        const interval = setInterval(() => {
+          MetricStatBySiteSubscription({
+            key: sKey,
+            siteId: site.id,
+            userId: user.id,
+            url: env.METRIC_URL,
+            orgName: user.orgName,
+            type: Stats_Type.Site,
+            from: getUnixTime() - 30,
+          });
+        }, 30000); // Update every 30 seconds
+
+        // Store interval ID for cleanup
+        return () => {
+          clearInterval(interval);
+          PubSub.unsubscribe(sKey);
+        };
+      });
+    };
```

Comment:
I'm not getting the reasoning behind this. Why we need period call's to a subscription?